### DISCUSSION TITLE
Port of Telescope-project

### DISCUSSION THREAD

**Discussion by dapc11** [ibhagwan/fzf-lua#1181] [Show and tell]
Thanks for a great plugin. Since transitioning away from Telescope, I've been missing https://github.com/nvim-telescope/telescope-project.nvim. So I have ported, to me, the most crucial parts of it. Hope it could be of help to someone else!

```lua
function()
  local fzf = require("fzf-lua")
  fzf.fzf_exec("fd '.git$' --prune -utd <repo dir 1> .. <repo dir n> | xargs dirname", {
    actions = {
      ["default"] = function(selected, _)
        vim.cmd("cd " .. selected[1])
        fzf.git_files()
      end,
      ["ctrl-f"] = function(selected, _)
        vim.cmd("cd " .. selected[1])
        fzf.grep_project()
      end,
      ["ctrl-r"] = function(selected, _)
        vim.cmd("cd " .. selected[1])
        fzf.oldfiles({ cwd = selected[1] })
      end,
    },
  })
end
```

**Comment by ibhagwan** [comment]
Near solution @dapc11.

- I edited your post to add lua syntax coloring to the code, you do this by adding `lua` after the 3 backticks starting the code block.
- `fzf.grep_project({ search = "" })`, the empty search part isn’t required, `grep_project` already is an alias to `grep({ search = '' })`

https://github.com/ibhagwan/fzf-lua/blob/main/OPTIONS.md#grep_project

https://github.com/ibhagwan/fzf-lua/blob/64f6eff4702c23c4de5320ed668343af1e4d679e/lua/fzf-lua/providers/grep.lua#L373-L385

**Reply by dapc11** [reply]
Thanks for review. I've edited the end result, see first post.

**Reply by diocletiann** [reply]
This maybe what I've been looking for, thanks! Any tips on how I could modify LazyVim's `project.nvim` config to include this?
```lua
{
  "ahmedkhalf/project.nvim",
  opts = {
    manual_mode = true,
  },
  event = "VeryLazy",
  config = function(_, opts)
    require("project_nvim").setup(opts)
    LazyVim.on_load("telescope.nvim", function()
      require("telescope").load_extension("projects")
    end)
  end,
  keys = {
    { "<leader>fp", "<Cmd>Telescope projects<CR>", desc = "Projects" },
  },
}
```

**Reply by dapc11** [reply]
Glad you like it! I think my initial snippet covers the most basic use case of project.nvim. I removed the dependency of both telescope and project.nvim (forgot to include this in the thread) when I introduced this solution to my config.

**Reply by diocletiann** [reply]
I see, makes sense. I'm still new to Neovim, where should I add your function once I disable telescope and project.nvim?

**Reply by dapc11** [reply]
No worries. Please skim through my dotfiles: https://github.com/dapc11/dnvim, more precisely, _lua/plugins/fzf.lua_ and the common files.
